index.bs

```diff
@@ -2156,10 +2187,117 @@ Base64-encoded string.
 
 TODO: Full page screenshots, and multiple output formats
 
+<div algorithm>
+To <dfn>normalise rect</dfn> given |rect|:
+
+Note: This ensures that the resulting rect has positive [=width dimension=] and
+[=height dimension=].
+
+1. Let |x| be |rect|'s [=x coordinate=].
+
+1. Let |width| be |rect|'s [=width dimension=].
+
+1. Let |y| be |rect|'s [=y coordinate=].
+
+1. Let |height| be |rect|'s [=height dimension=].
+
+1. If |width| is less than 0, set |x| to |x| - |width| and then set |width| to
```

Comment:
The width and height cannot be smaller than `0` given that both are of type `js-uint`.

Reply:
Note to self: those types are wrong. And we probably need to round to pixels after scaling when we actually render.

Reply:
I assume that you are still planning to do an update for that? I'll defer the next review until you closed this issue.

Reply:
Update now.